taxonomy-editor-frontend/src/pages/export/index.jsx

```diff
@@ -0,0 +1,145 @@
+import { Box, Grid, Typography, Alert, Snackbar, Button, Link as MuiLink, CircularProgress} from "@mui/material";
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import DialogTitle from '@mui/material/DialogTitle';
+import { useParams } from "react-router-dom";
+import { createBaseURL } from "../editentry/createURL";
+import { useState, useEffect } from "react";
+
+const ExportTaxonomy = ({setDisplayedPages}) => {
+    const { taxonomyName, branchName } = useParams();
+    const urlPrefix = `${taxonomyName}/${branchName}/`;
+    const baseURL = createBaseURL(taxonomyName, branchName);
+
+    const [loadingForDownload, setLoadingForDownload] = useState(false);
+    const [loadingForGithub, setLoadingForGitHub] = useState(false);
+    const [openSuccessDialog, setOpenSuccessDialog] = useState(false);
+    const [pullRequestURL, setPullRequestURL] = useState(null);
+    const [openErrorSnackbar, setOpenErrorSnackbar] = useState(false);
+    const [errorMessage, setErrorMessage] = useState("");
+
+    // Set url prefix for navbar component
+    useEffect(() => {
+        setDisplayedPages([
+            { url: urlPrefix+"entry", translationKey: "Nodes" },
+            { url: urlPrefix+"search", translationKey: "Search" },
+            { url: urlPrefix+"export", translationKey: "Export" }
+        ])
+    }, [urlPrefix, setDisplayedPages])
+
+    function handleDownload() {
+        setLoadingForDownload(true);
+        fetch(baseURL+'downloadexport', {
+            method : 'GET'
+        }).then((response) => {
+            return response.blob();
+        }).then((blob) => {
+            // Download taxonomy
+            var url = window.URL.createObjectURL(blob);
+            var a = document.createElement('a');
+            a.href = url;
+            a.download = taxonomyName + '.txt'
+            document.body.appendChild(a);
+            a.click(); a.remove();
```

Comment:
Ah, `window.open()` opens the text file in the browser, instead of downloading it. So, we download the text file as a blob and use a virtual link and click for downloading it.

Reply:
does it works without adding it to the DOM? Also, when using `window.URL.createObjectURL` you need to revoke: `URL.revokeObjectURL()` otherwise it stays in memory

Reply:
Yep, I'm removing it using the `URL.revokeObjectURL()` after the virtual click.